src/applications/disability-benefits/2346/components/ReviewPageSupplies.jsx

```diff
@@ -6,60 +6,107 @@ const ReviewPageSupplies = ({
+          You can't add batteries to your order at this time due to
+          ineligibility.
+        </p>
+      )}
+      {eligibility?.batteries &&
+        selectedBatteryProductInfo?.length <= 0 && (
+          <div className="empty-state-eligible-battery-text">
+            <p className="vads-u-font-style--italic vads-u-display--inline-block vads-u-margin-right--0p5">
+              No batteries added.
+            </p>
+            <button
+              className="va-button-link"
+              onClick={() => formContext.onEdit()}
+            >
+              Add batteries to your order
+            </button>
+          </div>
+        )}
     </dt>
-    <div className="vads-u-margin-bottom--3">
-      {selectedBatteryProductInfo &&
-        selectedBatteryProductInfo.map((product, index) => (
-          <dd
-            key={`${product.productId}_${index}`}
-            className="vads-u-background-color--gray-light-alt vads-u-width--full vads-u-padding--4 vads-u-display--flex vads-u-flex-direction--column vads-u-margin-bottom--1p5"
-          >
-            <h5 className="vads-u-font-size--h4 vads-u-margin-top--0">
-              {product.deviceName}
-            </h5>
-            <span>
-              {product.productName} batteries (Quantity: {product.quantity})
-            </span>
-          </dd>
-        ))}
-    </div>
+    {eligibility?.batteries && (
+      <div className="vads-u-margin-bottom--3">
+        {selectedBatteryProductInfo &&
+          selectedBatteryProductInfo.map((product, index) => (
+            <dd
+              key={`${product.productId}_${index}`}
+              className="vads-u-background-color--gray-light-alt vads-u-width--full vads-u-padding--4 vads-u-display--flex vads-u-flex-direction--column vads-u-margin-bottom--1p5"
+            >
+              <h5 className="vads-u-font-size--h4 vads-u-margin-top--0">
+                {product.deviceName}
+              </h5>
+              <span>
+                {product.productName} batteries (Quantity: {product.quantity})
+              </span>
+            </dd>
+          ))}
+      </div>
+    )}
     <dd className="vads-u-display--flex vads-u-flex-direction--column vads-u-margin-bottom--2">
       <span className="vads-u-font-weight--bold">
         You have requested to receive the following accessories:
       </span>
       <span>
-        ({selectedAccessoryProductInfo?.length || 0} out of{' '}
-        {accessorySupplies?.length || 0} selected)
+        ({selectedAccessoryProductInfo?.length} out of{' '}
+        {eligibility?.accessories ? accessorySupplies?.length : 0} selected)
       </span>
+      {!eligibility?.accessories && (
+        <p className="vads-u-font-style--italic empty-state-ineligible-accessory-text">
+          You can't add accessories to your order at this time due to
+          ineligibility.
+        </p>
+      )}
+      {eligibility?.accessories &&
+        selectedAccessoryProductInfo?.length <= 0 && (
```

Comment:
```<= 0``` on a length shouldn't happen because array length can never be negative. Not a blocker, though.

Reply:
Wasn't sure if it could be, thanks for confirming. I removed the catch for negative numbers.